src/core/common/settings.cpp

```diff
@@ -345,6 +351,13 @@ ThreadError otPlatSettingsGet(otInstance *aInstance, uint16_t aKey, int aIndex,
 
 ThreadError otPlatSettingsSet(otInstance *aInstance, uint16_t aKey, const uint8_t *aValue, int aValueLength)
 {
+    int length;
+
+    if (otPlatSettingsGet(aInstance, aKey, 0, NULL, &length) == kThreadError_None)
+    {
+        otPlatSettingsDelete(aInstance, aKey, 0);
```

Comment:
We should not be deleting before writing.  For example, if the device resets after the delete but before the write, there will be no value for this setting after reboot.

We should not need to delete.  We just need to indicate that this is the first item in the list of values for this key.